management/virtualmachine/entities.go

```diff
@@ -11,132 +12,235 @@ type VirtualMachineClient struct {
-	SSH                              SSH            `xml:",omitempty"`
-	CustomData                       string         `xml:",omitempty"`
+// Specifies the protocol and certificate information for a WinRM listener.
+type WinRMListener struct {
+	Protocol              WinRMProtocol // Specifies the protocol of listener.
+	CertificateThumbprint string        `xml:",omitempty"` // Specifies the certificate thumbprint for the secure connection. If this value is not specified, a self-signed certificate is generated and used for the Virtual Machine.
 }
 
+type WinRMProtocol string // see "WinRMProtocol*"" constants
+
+const (
+	// Enum values for WinRMProtocol
+	WinRMProtocolHttp  = WinRMProtocol("Http")
+	WinRMProtocolHttps = WinRMProtocol("Https")
+)
+
+// Specifies the SSH public keys and key pairs to use with the Virtual Machine.
 type SSH struct {
-	PublicKeys PublicKeyList
+	PublicKeys *[]PublicKey `xml:"PublicKeys>PublicKey"`
+	KeyPairs   *[]KeyPair   `xml:"KeyPairs>KeyPair"`
 }
 
-type PublicKeyList struct {
-	PublicKey []PublicKey
+// Specifies a public SSH key.
+type PublicKey struct {
+	Fingerprint string // Specifies the SHA1 fingerprint of an X509 certificate associated with the cloud service and includes the SSH public key.
+	Path        string // Usually /home/username/.ssh/authorized_keys
+	// Specifies the full path of a file, on the Virtual Machine, where the SSH public key is stored. If
+	// the file already exists, the specified key is appended to the file.
 }
 
-type PublicKey struct {
-	Fingerprint string
-	Path        string
+// Specifies an SSH keypair.
+type KeyPair struct {
+	Fingerprint string // Specifies the SHA1 fingerprint of an X509 certificate that is associated with the cloud service and includes the SSH keypair.
+	Path        string // Usually /home/username/.ssh/id_rsa
+	// Specifies the full path of a file, on the virtual machine, which stores the SSH private key. The
+	// file is overwritten when multiple keys are written to it. The SSH public key is stored in the same
+	// directory and has the same name as the private key file with .pub suffix.
 }
 
+// Specifies the properties that define an external endpoint for the Virtual Machine.
 type InputEndpoint struct {
-	LocalPort int
-	Name      string
-	Port      int
-	Protocol  string
-	Vip       string
+	LocalPort int                   // Specifies the internal port on which the Virtual Machine is listening.
+	Name      string                // Specifies the name of the external endpoint.
+	Port      int                   // Specifies the external port to use for the endpoint.
+	Protocol  InputEndpointProtocol //Specifies the transport protocol for the endpoint.
+	Vip       string                `xml:",omitempty"`
+}
+
+type InputEndpointProtocol string // see "InputEndpointProtocol*"" constants
+
+const (
+	// Enum values for InputEndpointProtocol
+	InputEndpointProtocolTcp = InputEndpointProtocol("TCP")
```

Comment:
Move the type next to identifier, type conversion will be inferred automatically